### PR TITLE
JButtWeld

### PR DESCRIPTION
**Pull request by emagdalenaC2i**

### CODE REVIEW

**Comment by emagdalenaC2i** [comment]
@mbartlett21 I have just added a commit with the oppositeDirection

File: weld.fs.js [hunk 1]
```diff
@@ -1314,70 +1314,107 @@ function sketchUButtWeld(context is Context, definition is map, thickness is Val
...
+    var distOut = (radius * cos(-angle)) + (distBase * tan(angle));
+    if (rootGap)
+    {
+        distOut = (radius * cos(-angle)) + (distBase - rootGapHeight) * tan(angle) + rootGapWidth;
+    }
+            
+    var oppDir = 1;
+    if (side2 && definition.oppositeDirection2 || !side2 && definition.oppositeDirection)
+    {
+        oppDir = -1 * oppDir;
```
Comment (mbartlett21) [inline weld.fs.js]: Can you use
```suggestion
        oppDir *= -1;
```

File: weld.fs.js [hunk 2]
```diff
@@ -1314,70 +1314,107 @@ function sketchUButtWeld(context is Context, definition is map, thickness is Val
...
-                    "end" : vector(-rootGapWidth / 2, -thickness + rootGapHeight)
+        skArc(profileSketch, "bottomArc1", {
+                    "start" : vector(oppDir * rootGapWidth, -thickness + rootGapHeight),
+                    "mid" : vector((oppDir * rootGapWidth) + (oppDir * radius * cos(ang1)), -thickness + radius + (radius * sin(ang1)) + rootGapHeight),
+                    "end" : vector((oppDir * rootGapWidth) + (oppDir * radius * cos(-angle)), -thickness + radius + (radius * sin(-angle)) + rootGapHeight)
                 });
         skLineSegment(profileSketch, "sideLine2", {
-                    "start" : vector(distOut, 0 * meter),
-                    "end" : vector(rootGapWidth / 2, -thickness + rootGapHeight)
+                    "start" : vector((oppDir * rootGapWidth) + (oppDir * radius * cos(-angle)), -thickness + radius + (radius * sin(-angle)) + rootGapHeight),
```
Comment (mbartlett21) [inline weld.fs.js]:
```suggestion
                    "start" : vector(oppDir * (rootGapWidth + radius * cos(-angle)), -thickness + radius + (radius * sin(-angle)) + rootGapHeight),
```

File: weld.fs.js [hunk 3]
```diff
@@ -1314,70 +1314,107 @@ function sketchUButtWeld(context is Context, definition is map, thickness is Val
...
+                    "end" : vector(distOut, 0 * meter)
+                });
+    }
     else
+    {
         skArc(profileSketch, "topLine", {
-                    "start" : vector(-distOut, 0 * meter),
-                    "mid" : vector(0 * meter, distOut / 5),
+                    "start" : vector(0 * meter, 0 * meter),
+                    "mid" : vector(distOut / 2.0, -offset),
```
Comment (mbartlett21) [inline weld.fs.js]:
```suggestion
                    "mid" : vector(distOut / 2, -offset),
```

File: weld.fs.js [hunk 4]
```diff
@@ -1314,70 +1314,107 @@ function sketchUButtWeld(context is Context, definition is map, thickness is Val
...
+        shape = definition.buttShape2;
+        offset = definition.buttOffset2;
+        radius = definition.buttRadius2;
+        angle = definition.buttAngle2;
+        rootGap = definition.buttRootGap2;
+        rootGapWidth = definition.buttRootGapWidth2;
+        rootGapHeight = definition.buttRootGapHeight2;
+    }
+            
+    var ang1 = (-angle - (((90 * degree) - angle) / 2.0));
```
Comment (mbartlett21) [inline weld.fs.js]:
```suggestion
    var ang1 = -angle/2 - 45 * degree;
```

File: weld.fs.js [hunk 5]
```diff
@@ -1314,70 +1314,107 @@ function sketchUButtWeld(context is Context, definition is map, thickness is Val
...
+                    "start" : vector(0 * meter, 0 * meter),
                     "end" : vector(distOut, 0 * meter)
                 });
-        // Si la forma de la soldadura No es plana, es convexa, crea un arco
+    }
+    else if (shape == WeldShape.CONVEX)
+    {
+        skArc(profileSketch, "topLine", {
+                    "start" : vector(0 * meter, 0 * meter),
+                    "mid" : vector(distOut / 2.0, offset),
```
Comment (mbartlett21) [inline weld.fs.js]:
```suggestion
                    "mid" : vector(distOut / 2, offset),
```

File: weld.fs.js [hunk 6]
```diff
@@ -1314,70 +1314,107 @@ function sketchUButtWeld(context is Context, definition is map, thickness is Val
...
+        oppDir = -1 * oppDir;
+        distOut = -distOut;
+    }
 
-    // Si la forma de la soldadura es plana crea una linea
     if (shape == WeldShape.FLAT)
+    {
         skLineSegment(profileSketch, "topLine", {
-                    "start" : vector(-distOut, 0 * meter),
+                    "start" : vector(0 * meter, 0 * meter),
```
Comment (mbartlett21) [inline weld.fs.js]:
```suggestion
                    "start" : vector(0, 0) * meter,
```

**Comment by emagdalenaC2i** [comment]
@mbartlett21 done

**Pull request merged**